sherpa/astro/io/pyfits_backend.py

```diff
@@ -1163,23 +1179,29 @@ def set_arrays(filename, args, fields=None, ascii=True, clobber=False):
         if len(arg) != size:
             raise IOErr('arraysnoteq')
 
-    if not ascii and fields is None:
-        fields = [f'col{ii + 1}' for ii in range(len(args))]
-
     if fields is not None and len(args) != len(fields):
         raise IOErr("wrongnumcols", len(args), len(fields))
 
     if ascii:
-        write_arrays(filename, args, fields, clobber=clobber)
+        # Historically, the serialization doesn't quite match the
+        # AstroPy Table version, so stay with write_arrays. We do
```

Comment:
Astropy table surely has options to make the format we want (I promise!) but since we have `write_tables` already, we might as well keep it here.

Reply:
Yeah - I am trying to not significantly change things at this time. For some definition of "significant".

One thing I spent time on when adding tests to the I/O code a while ago was to try and ensure we had "round trip" tests - e.g. that we can read back in a file we create. The idea is that the output text may change slightly (e.g. the column headings) but conceptually it's the same data. Or, at least, that's the plan.